numpy/typing/tests/data/reveal/ctypeslib.pyi

```diff
@@ -80,16 +80,17 @@ assert_type(np.ctypeslib.as_array(1), npt.NDArray[Any])
 assert_type(np.ctypeslib.as_array(pointer), npt.NDArray[Any])
 
 if sys.platform == "win32":
-    assert_type(np.ctypeslib.as_ctypes_type(np.int_), type[ct.c_int])
-    assert_type(np.ctypeslib.as_ctypes_type(np.uint), type[ct.c_uint])
-    assert_type(np.ctypeslib.as_ctypes(AR_uint), ct.Array[ct.c_uint])
-    assert_type(np.ctypeslib.as_ctypes(AR_int), ct.Array[ct.c_int])
-    assert_type(np.ctypeslib.as_ctypes(AR_uint.take(0)), ct.c_uint)
-    assert_type(np.ctypeslib.as_ctypes(AR_int.take(0)), ct.c_int)
+    # Mainly on windows int is the same size as long but gets picked first:
+    assert_type(np.ctypeslib.as_ctypes_type(np.long), type[ct.c_int])
+    assert_type(np.ctypeslib.as_ctypes_type(np.ulong), type[ct.c_uint])
+    assert_type(np.ctypeslib.as_ctypes(AR_ulong), ct.Array[ct.c_uint])
+    assert_type(np.ctypeslib.as_ctypes(AR_long), ct.Array[ct.c_int])
+    assert_type(np.ctypeslib.as_ctypes(AR_long.take(0)), ct.c_int)
+    assert_type(np.ctypeslib.as_ctypes(AR_ulong.take(0)), ct.c_uint)
 else:
-    assert_type(np.ctypeslib.as_ctypes_type(np.int_), type[ct.c_long])
-    assert_type(np.ctypeslib.as_ctypes_type(np.uint), type[ct.c_ulong])
-    assert_type(np.ctypeslib.as_ctypes(AR_uint), ct.Array[ct.c_ulong])
-    assert_type(np.ctypeslib.as_ctypes(AR_int), ct.Array[ct.c_long])
-    assert_type(np.ctypeslib.as_ctypes(AR_uint.take(0)), ct.c_ulong)
-    assert_type(np.ctypeslib.as_ctypes(AR_int.take(0)), ct.c_long)
+    assert_type(np.ctypeslib.as_ctypes_type(np.long), type[ct.c_long])
+    assert_type(np.ctypeslib.as_ctypes_type(np.ulong), type[ct.c_ulong])
+    assert_type(np.ctypeslib.as_ctypes(AR_ulong), ct.Array[ct.c_ulong])
+    assert_type(np.ctypeslib.as_ctypes(AR_long), ct.Array[ct.c_long])
+    assert_type(np.ctypeslib.as_ctypes(AR_long.take(0)), ct.c_long)
+    assert_type(np.ctypeslib.as_ctypes(AR_ulong.take(0)), ct.c_ulong)
```

Comment:
LGTM here; nice to see you managed to get things working without commenting them out!